### PR TITLE
LVRUBYM-342: Add groups resolve for main state

### PR DESCRIPTION
**Pull request by vikabelous**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVRUBYM-342)

## Code reviewers

- [x] @skrypnyk88  
- [x] @JulkaIII 
- [x] @Mykhailo123 

### Second Level Review

- [ ] @4asof4ik

## Summary of issue

Need to fix groupList request issue

## Summary of change

- removed group component and added group select to main template.
- added resolve for main state so that main state and it's child states are not loaded until groups promise is resolved.
- added watcher for currentGroupDay.group_id value to reload child state when group is changed

## Testing approach

No tests needed

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/js/common/services/group.service.js [hunk 1]
```diff
@@ -6,17 +6,18 @@ module.exports = angular
   ])
   .factory('groupService', groupService);
 
-groupService.$inject = ['groupResource'];
+groupService.$inject = ['groupResource', 'currentGroupDay'];
```
Comment (skrypnyk88) [inline app/js/common/services/group.service.js]: You should write code in one style. Please write 'current Group Day' in next line.

Comment (vikabelous) [inline app/js/common/services/group.service.js]: I think it's not necessary to write each dependency on new line if there are just few of them, it's easier to read.
[Style guides](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y091)

File: app/js/main/components/main.template.html [hunk 2]
```diff
@@ -6,7 +6,9 @@
 
     <div flex layout="row" layout-align="space-around center">
 
-      <group-list></group-list>
+      <md-select ng-model="$ctrl.currentGroupDay.group_id" >
+        <md-option ng-value="group.id" ng-repeat="group in $ctrl.groups">{{ group.name }}</md-option>
```
Comment (skrypnyk88) [inline app/js/main/components/main.template.html]: <md-option ng-value="group.id" ng-repeat="group in $ctrl.groups">
    {{ group.name }}
</md-option>

Comment (vikabelous) [inline app/js/main/components/main.template.html]: ?)

File: app/js/common/services/group.service.js (hunk 1 above)
Comment (skrypnyk88) [inline app/js/common/services/group.service.js]: You are right. But we have our group style.

**Pull request merged**